src/main/java/StudentManagementSystem.java

```diff
@@ -0,0 +1,55 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Scanner;
+
+public class StudentManagementSystem {
```

Comment:
I'd recommend using the Singleton pattern here. This way you'd solve the problem with passing down fields between different classes (like `systemState` for example) and make sure you have only one instance of this class.